src/main/java/org/myrobotlab/service/Runtime.java

```diff
@@ -3381,13 +3361,16 @@ static public void disableLogging() {
    */
   @Override
   public void releaseService() {
-    super.releaseService();
     if (runtime != null) {
+      runtime.purgeTasks();
+      runtime.stopService();
       runtime.stopInteractiveMode();
-
       runtime.getRepo().removeStatusPublishers();
+      registry = new TreeMap<>();

```

Comment:
This can be done with a `ConcurrentHashMap` and a call to `clear()`

Reply:
TreeMap is desired, so I made it threadsafe